### PR TITLE
Add support for name and version to bump release

### PR DESCRIPTION
**Pull request by lcarva**
The `bump_release` plugin uses koji's `getLatestBuilds` method which groups results by name only, not by name-version. This produces incorrect behavior when a tag contains multiple versions for same component.

Since we have both name (component) and version already from Dockerfile, just use `getNextRelease`.
This works because name and version have to be unique within koji, checking the target and package seems to be unnecessary.

My only concern is that we're changing that plugin's definition which was released today. Let's consider making target an optional parameter in the mean time?

### CODE REVIEW

**Comment by twaugh** [comment]
Great catch!

Yes, we need to keep target as an optional parameter and ignore it, at least for a short time.

File: atomic_reactor/plugins/pre_bump_release.py [hunk 1]
```diff
@@ -50,34 +48,26 @@ def run(self):
...
 
-        # No release labels are set so set them.
         component_label = get_preferred_label_key(dockerfile_labels,
                                                   'com.redhat.component')
         try:
             component = dockerfile_labels[component_label]
         except KeyError:
             raise RuntimeError("missing label: {}".format(component_label))
 
-        target_info = self.xmlrpc.getBuildTarget(self.target)
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_bump_release.py]: We shouldn't be removing those checks

Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: I don't think that logic belongs in this plugin. From this plugin's point of view a target is not required at all.

Comment (vrutkovs) [inline atomic_reactor/plugins/pre_bump_release.py]: Agreed, I'll prepare a PR to move this to ```pre_koji```

File: atomic_reactor/plugins/pre_bump_release.py [hunk 2]
```diff
@@ -23,18 +23,24 @@ class BumpReleasePlugin(PreBuildPlugin):
     key = "bump_release"
     is_allowed_to_fail = False  # We really want to stop the process
 
-    def __init__(self, tasker, workflow, target, hub):
+    # The target parameter is no longer used by this plugin. It's
+    # left as an optional parameter to allow a graceful transition
+    # in osbs-client.
+    def __init__(self, tasker, workflow, target=None, hub=None):
```
Comment (twaugh) [inline atomic_reactor/plugins/pre_bump_release.py]: Actually plugin parameters are always passed by keyword, so it's fine to swap target and hub around and only make target optional.

Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: Done.

File: tests/plugins/test_bump_release.py [hunk 3]
```diff
@@ -58,64 +61,69 @@ def test_component_missing(self, tmpdir):
...
-        ('comp', 'dest_tag', False),
-        ('comp', 'wrong_tag', True),
-        ('wrong_comp', 'dest_tag', True),
+    @pytest.mark.parametrize('version', [
+        {'version': '7.1'},
+        {'Version': '7.2'},
     ])
-    def test_increment(self, tmpdir, release_label, latest_builds, next_release,
-                       expected, component, tag, throws_exception):
+    @pytest.mark.parametrize('include_target', [
```
Comment (twaugh) [inline tests/plugins/test_bump_release.py]: Nice :)

**Comment by twaugh** [comment]
LGTM

**Comment by vrutkovs** [comment]
LGTM

**Pull request merged**